contrib/repro_test.bzl

```diff
@@ -0,0 +1,243 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test rule for verifying a given container_image target produces a
+   deterministic Docker image.
+
+This rule builds the given container_image twice in isolation to determine if
+the given image is deterministic.
+It does so by mounting the provided Bazel project files into the image to build
```

Comment:
into the 'base' image. Next it starts two containers and builds the supplied container_image target
(being `test_image`) inside each of the containers. The rule then extracts the output ..

Reply:
Note that base_image must have Bazel +docker installed and in the path

Reply:
Done